packages/typespec-python/src/emitter.ts

```diff
@@ -172,6 +172,13 @@ function handleDiscriminator(context: SdkContext, type: Model, model: Record<str
                 type: discriminatorType,
             };
             propertyCopy.description = "";
+            for (let i = 0; i < model.properties.length; i++) {
+                if (model.properties[i].wireName === discriminator.propertyName) {
+                    model.properties[i] = propertyCopy;
+                    return;
+                }
+            }
+
```

Comment:
Without the change, there will be [duplicated discriminator](https://github.com/Azure/autorest.python/blob/925d7df3f046e7e6f061749bd5bb49033878b18d/packages/typespec-python/test/generated/typetest-model-nesteddiscriminator/typetest/model/nesteddiscriminator/models/_models.py#L88-L91C1) for [scenario](https://github.com/Azure/cadl-ranch/blob/8df197246599f76cb614d08248ec5f31d112c943/packages/cadl-ranch-specs/http/type/model/inheritance/nested-discriminator/main.tsp#L16-L21)